test/parallel/test-fs-promises-writefile.js

```diff
@@ -35,7 +35,14 @@ async function doRead() {
   assert.deepStrictEqual(buf, data);
 }
 
+async function doReadWithEncoding() {
+  const data = await fsPromises.readFile(dest, 'utf-8');
+  const buf = fs.readFileSync(dest, 'utf-8');
+  assert.deepStrictEqual(buf, data);
```

Comment:
Can we assert the type of `data` and `buf` as well, here?

Reply:
Sure